src/components/configurable_session.rs

```diff
@@ -0,0 +1,360 @@
+                    })
+                })
+            }).create_action_row(|r|{
+                r.create_select_menu(|s| {
+                    s.custom_id("configurable_session:prettify").placeholder("Prettify output").min_values(1).max_values(1).options(|o| {
+                        o.create_option(|o| o.default_selection(config.pretty).label("Pretty output").value("true"))
+                         .create_option(|o| o.default_selection(!config.pretty).label("Raw output").value("false"))
+                    })
+                })
+            }).create_action_row(|r|{
+                r.create_select_menu(|s| {
+                    s.custom_id("configurable_session:require_query").placeholder("Require /query").min_values(1).max_values(1).options(|o| {
+                        o.create_option(|o| o.default_selection(!matches!(conn, ConnType::EphemeralChannel)).label("Must use /query or /q").value("true"))
+                         .create_option(|o| o.default_selection(matches!(conn, ConnType::EphemeralChannel)).label("All messages are queries").value("false"))
+                    })
+                })
+            }).create_action_row(|mut r| {
+                r = r.create_button(|b| b.custom_id("configurable_session:big_query").label("Big Query").style(Primary).emoji('📝'));
+                r = r.create_button(|b| b.custom_id("configurable_session:export").label("Export").style(Success).emoji('📃'));
+                if matches!(conn, ConnType::Thread) {
+                    r = r.create_button(|b| b.custom_id("configurable_session:rename_thread").label("Rename thread").style(Secondary).emoji(TryInto::<ReactionType>::try_into("✏️").expect("Failed to convert emoji")));
+                }
+                r.create_button(|b| b.custom_id("configurable_session:stop").label("Stop session").style(Danger).emoji('💣'))
+            })
+        })
+    }).await?;
+
+    Ok(())
+}
+
+#[instrument(skip(ctx, event))]
+pub async fn handle_component(
+    ctx: &Context,
+    event: &MessageComponentInteraction,
+    channel: &ChannelId,
+    id: &str,
+    values: &[String],
+) -> Result<()> {
+    let db_exists = DBCONNS.lock().await.get(&channel.0).is_some();
+
+    match (id, db_exists) {
+        ("format", true) | ("prettify", true) | ("require_query", true) => {
+            debug!("Updating config");
+            let mut db = DBCONNS
+                .lock()
+                .await
+                .get_mut(&channel.0)
+                .expect("DB disappeared between now above check")
+                .clone();
+            match id {
+                "format" => db.json = values[0] == "json",
+                "prettify" => db.pretty = values[0] == "true",
+                "require_query" => db.require_query = values[0] == "true",
+                _ => unreachable!(),
+            }
+            DBCONNS.lock().await.insert(channel.0, db);
```

Comment:
if you did get_mut you shouldn't need to insert as well. Or if you are going to clone then you don't need get_mut